src/AppBundle/Service/CsvGeneratorService.php

```diff
@@ -62,6 +62,33 @@ public function generateTransactionsCsv(ReportInterface $report)
         return $csvContent;
     }
 
+    /**
+     * Generate all the report submissions as csv
+     *
+     * @param array $records
+     *
+     * @return string Csv content
+     */
+    public function generateReportSubmissionsCsv($records)
+    {
+        $this->logger->info('Generating Report submissions CSV : ');
+
+        // initialize temporary fp
+        $this->fd = fopen('php://temp/maxmemory:1048576', 'w');
+        if ($this->fd === false) {
+            $this->logger->error('Failed to open Temporary file');
+            die('Failed to open temporary file');
```

Comment:
exception ?

Reply:
yep, moved to new private method and throw new Exception